packages/js/src/utils/package-json/update-package-json.ts

```diff
@@ -44,45 +46,68 @@ export function updatePackageJson(
   target: ProjectGraphProjectNode,
   dependencies: DependentBuildableProjectNode[]
 ): void {
-  const pathToPackageJson = join(
-    context.root,
-    options.projectRoot,
-    'package.json'
+  const packageJson = createPackageJson(
+    context.projectName,
+    context.projectGraph,
+    {
+      root: context.root,
+      // By default we remove devDependencies since this is a production build.
+      isProduction: true,
+    }
   );
-
-  const packageJson = fileExists(pathToPackageJson)
-    ? readJsonFile(pathToPackageJson)
-    : { name: context.projectName };
-
+  // make custom modifications to package.json
+  updatePackageJsonContent(packageJson, options);
   if (options.excludeLibsInPackageJson) {
     dependencies = dependencies.filter((dep) => dep.node.type !== 'lib');
   }
+  if (options.updateBuildableProjectDepsInPackageJson) {
+    addMissingLibDependencies(packageJson, context, dependencies);
```

Comment:
Since `createPackageJson` is doing most of the things `updateBuildableProjectPackageJsonDependencies` is doing, we only need to add existing lib dependencies to package json